pulser/pulse.py

```diff
@@ -87,8 +87,10 @@ def __post_init__(self) -> None:
                 "greater than or equal to zero."
             )
 
-        self.__dict__["phase"] %= 2 * np.pi
-        self.__dict__["post_phase_shift"] %= 2 * np.pi
+        old_phase = object.__getattribute__(self, "phase")
+        old_pps = object.__getattribute__(self, "post_phase_shift")
```

Comment:
I think getting the attributes normally (e.g. `self.phase`) is not blocked, so I think this is not necessary.